### ISSUE TITLE
Setup Jitsi Meet on Apache2

### ISSUE THREAD

**Issue by PhuCuong14ctt** [jitsi/jitsi-meet#2728]
I followed Quick Install Jitsi Meet. When I open the browser and connect to the server all I get is the apache2 default index.html while I want to get to jitsi web interface.

**chandrantwins** said:

> you have to change server root path.. inside /etc/apache2/sites-enabled/yourdomain.conf

What do i have to change?
Please help me check my file /etc/apache2/sites-enabled/yourdomain.conf. Did you mean server root path = DocumentRoot?

```
<VirtualHost *:80>
    ServerName jitsimeet
    Redirect permanent / https://jitsimeet
    RewriteEngine On
    RewriteCond %{HTTPS} off
    RewriteRule ^ https://%{HTTP_HOST}%{REQUEST_URI} [R=301,L]
</VirtualHost>
 
<VirtualHost *:443>
 
  ServerName jitsimeet
 
  SSLProtocol TLSv1 TLSv1.1 TLSv1.2
  SSLEngine on
  SSLProxyEngine on
  SSLCertificateFile /etc/jitsi/meet/jitsimeet.crt
  SSLCertificateKeyFile /etc/jitsi/meet/jitsimeet.key
  SSLCipherSuite "EECDH+ECDSA+AESGCM:EECDH+aRSA+AESGCM:EECDH+ECDSA+SHA256:EECDH+aRSA+SHA256:EECDH+ECDSA+SHA384:EECDH+ECDSA+SHA256:EECDH+aRSA+SHA384:EDH+aRSA+AESGCM:EDH+aRSA+SHA256:EDH+aRSA:EECDH:!aNULL:!eNULL:!MEDIUM:!LOW:!3DES:!MD5:!EXP:!PSK:!SRP:!DSS:!RC4:!SEED"
  SSLHonorCipherOrder on
  Header set Strict-Transport-Security "max-age=31536000"
 
  DocumentRoot "/usr/share/jitsi-meet"
  <Directory "/usr/share/jitsi-meet">
    Options Indexes MultiViews Includes FollowSymLinks
    AddOutputFilter Includes html
    AllowOverride All
    Order allow,deny
    Allow from all
  </Directory>
 
  ErrorDocument 404 /static/404.html
 
  Alias "/config.js" "/etc/jitsi/meet/jitsimeet-config.js"
  <Location /config.js>
    Require all granted
  </Location>
 
  ProxyPreserveHost on
  ProxyPass /http-bind http://localhost:5280/http-bind/
  ProxyPassReverse /http-bind http://localhost:5280/http-bind/
 
  RewriteEngine on
  RewriteRule ^/([a-zA-Z0-9]+)$ /index.html
</VirtualHost>
```

**Comment by chandrantwins** [comment]
have you checked whether this location have /usr/share/jitsi-meet folder and all files

**Comment by chandrantwins** [comment]
just follow this steps.. you will succeed
https://github.com/jitsi/jitsi-meet/issues/2531

**Comment by PhuCuong14ctt** [comment]
Thanks, i will try.

**Comment by damencho** [comment]
@PhuCuong14ctt mind that tutorial is running jitsi-meet from source, you will not need everything from there.

**Comment by PhuCuong14ctt** [comment]
Yes, i know it. I saw that tutorial before **chandrantwins** had recommended but now i really got stuck.  **damencho** 🥇 can you show me a better solution? I don't know why other people can install jitsi easily. Maybe God is challenging me 😢

**Comment by damencho** [comment]
I'm not sure what can be the problem. We already have in package installation detection for apache2 existence. 
You can check your apache logs for errors and see does it complain something about the jitsi-meet virtual hosts.

If you are willing to go on the path of cleaning everything and starting all over again, here are some steps to try:
So make sure you have apache2 installed `dpkg -l | grep apache`, make sure nginx is not installed `dpkg -l | grep nginx`.
Then clean everything jitsi-meet specific, this will clean any custom settings you have:
```
apt-get --purge remove jitsi-meet jitsi-meet-web-config jitsi-meet-prosody jitsi-meet-web jicofo jitsi-videobridge prosody
rm -rf /etc/jitsi
apt-get install jitsi-meet
```
This will install jitsi-meet and all the components and configure apache2 to use it, it should work without a problem. If you still cannot load jitsi-meet it is some other apache2 configuration conflicting with the virtual hosts we create, you need to check your log and error log of apache, does it complain for something on reloading.

**Comment by PhuCuong14ctt** [comment]
I chose cleaning everything.
I checked apache log in file /var/log/apche2/error but i didn't see the problem here.
Can you teamviewer to help me? I think that's the final solution for me. Please damencho God 😄

**Comment by damencho** [comment]
Sorry, I have no time to do that. 
Is there anything for your virtual host in apache logs? What do they say on restarting?

**Comment by PhuCuong14ctt** [comment]
> [Thu Apr 05 02:08:30.473704 2018] [ssl:warn] [pid 5599:tid 140347621345152] AH01906: jitsimeet:443:0 server certificate is a CA certificate (BasicConstraints: CA == TRUE !?)
> [Thu Apr 05 02:08:30.477640 2018] [ssl:warn] [pid 5599:tid 140347621345152] AH01909: jitsimeet.southeastasia.cloudapp.azure.com:443:0 server certificate does NOT include an ID which matches the server name
> [Thu Apr 05 02:08:30.477795 2018] [mpm_event:notice] [pid 5599:tid 140347621345152] AH00489: Apache/2.4.18 (Ubuntu) OpenSSL/1.0.2g configured -- resuming normal operations
> [Thu Apr 05 02:08:30.477807 2018] [core:notice] [pid 5599:tid 140347621345152] AH00094: Command line: '/usr/sbin/apache2'

**Comment by damencho** [comment]
Maybe it is some azure configuration where you forward/open ports to the machine.

**Comment by PhuCuong14ctt** [comment]
I thought the same as you so i config it already

[](https://ibb.co/ewvaCH)

**Comment by damencho** [comment]
That is port 43 not 443. You also need to forward and udp 10000 and tcp 4443.

**Comment by PhuCuong14ctt** [comment]
In the end, i know what my real problem is.

When jitsi ask me hostname, i should input dns name of azure instead of hostname that i created in /etc/hosts
I run jitsi-meet successfully but i don't know why jitsi don't auto forward me from http to https?

[Port Image](https://ibb.co/c8hfWc)

**Comment by damencho** [comment]
This is if jvb is not running on a machine with public address. You need to forward tcp 4443 and udp 10000 to the jvb machine and enable these props.

**Comment by PhuCuong14ctt** [comment]
I'm a newbie with Linux.
How to 
> forward tcp 4443 and udp 10000 to the jvb machine and enable these props.

I tried socat
```
socat TCP-LISTEN:80,fork UDP:20.188.96.200:10000
socat TCP-LISTEN:80,fork TCP:20.188.96.200:4443
```
and i failed

**Comment by saghul** [comment]
If you machine is running behind NAT, you need to forward those ports in your router.

**Comment by PhuCuong14ctt** [comment]
I'm done.
Jitsi is so awesome. All of errors is my fault. S*** me 👎

Thanks for all reponses. @saghul, @chandrantwins , @damencho god.
Help me close this issue.

**Comment by 404Prachi** [comment]
I am facing same issue. I see apache2 index page not jitsi home page ....:(

**Comment by facb69** [comment]
Sorry to comment on this old post but I think that I have something to add.
I tried to install Jitsi on my home server. I used all the official instructions and after the installation was finished, I couldn't find the virtualhost and the URL opened apache index.html.
For my surprise, Nginx was also installed on the server. It was not active, only installed so, the installation process "saw" Nginx and configured it accordingly.
So I removed nginx and "purge remove" everyting Jitsi like @damencho explained. The installation now worked flawlessly.

Tips befor installing Jitsi: 
1. Check if there's any application using the ports 3478, 5349, 4443 and 10000 (this one is used by Webmin);
2. Check if you both Nginx and Apache and choose wich you want to use.

Cheers.
